data-document-model/src/main/java/io/micronaut/data/document/model/query/builder/CosmosSqlQueryBuilder.java

```diff
@@ -99,6 +99,11 @@ protected void appendProjectionRowCountDistinct(StringBuilder queryString, Query
         throw new UnsupportedOperationException("Count distinct is not supported by Micronaut Data Azure Cosmos.");
     }
 
+    @Override
+    public boolean supportsCountDistinct() {
+        return false;
```

Comment:
This will not use distinct as not supported in query building but won't have issues like in other SQL dialects because joins in Cosmos are different than regular SQL dialects.